forge-gui/res/cardsfolder/upcoming/fortified_beachhead.txt

```diff
@@ -0,0 +1,13 @@
+Name:Fortified Beachhead
+ManaCost:no cost
+Types:Land
+K:ETBReplacement:Other:DBReveal
+SVar:DBReveal:DB$ Reveal | RevealValid$ Soldier | Optional$ True | RememberRevealed$ True | SubAbility$ DBTap | SpellDescription$ As CARDNAME enters the battlefield, you may reveal a Soldier card from your hand. CARDNAME enters the battlefield tapped unless you revealed a Soldier card this way or you control a Soldier.
+SVar:DBTap:DB$ Tap | ETB$ True | Defined$ Self | ConditionCheckSVar$ SoldierPresence | ConditionSVarCompare$ EQ0 | SubAbility$ DBCleanup
```

Comment:
What about an UnlessCost on this in addition to a IsPresent EQ0? Then you could eliminate lines 5, 7-9

Reply:
I've copied this from Temple of the Dragon Queen, I guess that can be simplified as well

Reply:
This doesn't work because the land will enter tapped if you have a soldier on the field but don't reveal something

Reply:
![image](https://user-images.githubusercontent.com/67333662/199527810-585a5203-d5bf-41c1-90b4-ac234d81d878.png)